homeassistant/__main__.py

```diff
@@ -109,10 +109,71 @@ def get_arguments():
         '--open-ui',
         action='store_true',
         help='Open the webinterface in a browser')
+    parser.add_argument(
+        '-v', '--verbose',
+        action='store_true',
+        help="Enable verbose logging.")
+    parser.add_argument(
+        '--pid-file',
+        metavar='path_to_pid_file',
+        default=None,
+        help='Path to PID file useful for running as daemon')
+    parser.add_argument(
+        '--daemon',
+        action='store_true',
+        help='Run Home Assistant as daemon')
 
     return parser.parse_args()
 
 
+def _daemonize():
+    """ Move current process to daemon process """
+    # create first fork
+    pid = os.fork()
+    if pid > 0:
+        sys.exit(0)
+
+    # decouple fork
+    os.setsid()
+    os.umask(0)
+
+    # create second fork
+    pid = os.fork()
+    if pid > 0:
+        sys.exit(0)
+
+
+def setup_daemon(pid_file, run_as_daemon):
```

Comment:
I think that this function can be broken into three parts:

 1. kill existing HA if pid given
 2. Daemonize if requested
 3. Save pid if pid is given